.github/workflows/ci.yml

```diff
@@ -0,0 +1,36 @@
+on: [push, pull_request]
+
+jobs:
+  # This matrix job runs the test suite against multiple Ruby versions
+  test_matrix:
+    strategy:
+      fail-fast: false
+      matrix:
+        # Due to https://github.com/actions/runner/issues/849, we have to use quotes for '3.0'
+        ruby: [ 2.7, '3.0', 3.1 ]
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v2
+      - uses: ruby/setup-ruby@v1
+        with:
+          ruby-version: ${{ matrix.ruby }}
+          bundler-cache: true
+      - run: bundle exec rake
+
+  # Branch protection rules cannot directly depend on status checks from matrix jobs.
+  # So instead we define `test` as a dummy job which only runs after the preceding `test_matrix` checks have passed.
+  # Solution inspired by: https://github.community/t/status-check-for-a-matrix-jobs/127354/3
```

Comment:
I wonder if we want a shorter comment for this, if we're pasting it, since it could age badly. Not a biggie though and I don't think we've got a good thing to link to.

I'll just see about updating: https://docs.publishing.service.gov.uk/manual/test-and-build-a-project-with-github-actions.html later

Reply:
Going to leave this for now since I'm not sure what a less ageing-badly message would be, but happy to update if we settle on something better in future :-)